### PR TITLE
fix normalization of sky and time marginalization to reduce integral …

### PR DESCRIPTION
**Pull request by ahnitz**
…uncertainty

This reduces the uncertainty in the monte-carlo integral for time / sky marginalization. I was using a normalization that incorrectly introduced a large sample variance when in fact it was not needed. In testing this means that the likelihood (not log likelihood thankfully) std deviation is ~ 5-10% after 1000 samples and improves with root(N) as expected. For many cases, this is sufficient unless the level of accuracy required is very high. 

I've also added some comments for future improvements. The marginalizations implicitly assume a uniform time prior, which should be alleviated in a future pull request.

### CODE REVIEW

File: pycbc/inference/models/tools.py [hunk 1]
```diff
@@ -45,16 +45,16 @@ def draw_sample(loglr, size=None):
     cdf = numpy.exp(loglr).cumsum()
     cdf /= cdf[-1]
     xl = numpy.searchsorted(cdf, x)
+    
+    loglr = loglr - logsumexp(loglr)
```
Comment (WuShichao) [inline pycbc/inference/models/tools.py:49]: Thanks for the fix. What's the difference between normalization on `cdf` of `loglr` (previous code) and on `loglr` itself (this PR)? Can we remove the previous code? Here you generate `xl` twice.

File: pycbc/inference/models/tools.py [hunk 2]
```diff
@@ -120,7 +120,7 @@ def pop_prior(param):
         self.reconstruct_phase = False
         self.reconstruct_distance = False
         self.reconstruct_vector = False
-        self.precalc_antennna_factors = False
+        self.precalc_antenna_factors = False
```
Comment (WuShichao) [inline pycbc/inference/models/tools.py:123]: This typo has been fixed in another merged PR; please rebase it.

File: pycbc/inference/models/tools.py (hunk 1 above)
Comment (ahnitz) [reply pycbc/inference/models/tools.py:49]: I'll need to remove this. I accidentally left it in for debugging purposes. I wasn't quite sure where the problem was originally. It's not here, it was the fact that the prior normalization was tied to the samples rather than upfront. This caused a large uncertainty depending on how the samples ended up distributed.

File: pycbc/inference/models/tools.py [hunk 3]
```diff
@@ -363,7 +368,7 @@ def draw_times(self, snrs, size=None):
         starts = []
         ends = []
 
-        tmin, tmax = tcmin - dt, tcmax + dt
+        tmin, tmax = tcmin + dt - snrs[iref].delta_t, tcmax + dt + snrs[iref].delta_t
```
Comment (WuShichao) [inline pycbc/inference/models/tools.py:371]: Why did you change `tcmin - dt` to `tcmin + dt`? Is `snrs[iref].delta_t` the time resolution of the SNR time series for a detector?

**Comment by WuShichao** [comment]
Some code climate issues need to be fixed.

File: pycbc/inference/models/tools.py (hunk 3 above)
Comment (ahnitz) [reply pycbc/inference/models/tools.py]: To make sure the boundary condition was satisfied.

**Comment by ahnitz** [comment]
@WuShichao The remaining cc issues can't be resolved and were not introduced in this PR.
